controller/api/models.py

```diff
@@ -353,6 +367,58 @@ def _start_containers(self, to_add):
         if set([c.state for c in to_add]) != set(['up']):
             err = 'warning, some containers failed to start'
             log_event(self, err, logging.WARNING)
+        # if the user specified a health check, try checking to see if it's running
+        try:
+            config = self.config_set.latest()
+            if 'HEALTHCHECK_URL' in config.values.keys():
+                self._healthcheck(to_add, config.values)
+        except Config.DoesNotExist:
+            pass
+
+    def _healthcheck(self, containers, config):
+        # if at first it fails, back off and try again at 10%, 50% and 100% of INITIAL_DELAY
+        intervals = [1.0, 0.1, 0.5, 1.0]
+        # HACK (bacongobbler): we need to wait until publisher has a chance to publish each
```

Comment:
Unfortunately, there is a bug in go-etcd that causes the watch event to fire whenever a key is updated *or* a TTL of a key is updated. This causes a flood of requests to etcd, which has hurt us in the past. Hoping that using etcd2 and a fixed go-etcd library will rectify this, but for now I think we have to avoid using watch.

It'd be great if we could come up with another alternative than sleeping, but not sure if we can.